### DISCUSSION TITLE
Game of Life

### DISCUSSION THREAD

**Discussion by Paalui** [TodePond/CellPond#352] [Show and tell]
I made Game of Life
I had to use 66 rules but i think it would be possible to only use half if you subtract/add colors.
Because of the many rules it runs very slowly and even more slowly if the whole display is synchronized.

Is there a better way to do it?/Has anyone made a better version already?

[GameOfLifeSync.json](https://github.com/user-attachments/files/16045525/GameOfLifeSync.json)
[GameOfLifeAsync.json](https://github.com/user-attachments/files/16045523/GameOfLifeAsync.json)

[Video of sync version](https://youtu.be/P4EQz9UbPHo)
[Video of async version](https://youtu.be/JcG293k0o1Y)

**Comment by TodePond** [comment]
HOLY SHIT thats amazing. please can I post this up on social media? or you're welcome to and I'll retweet/boost/whatever

**Reply by TodePond** [reply]
If you have the raw video files, they're probably small enough to tweet/toot

**Reply by Paalui** [reply]
Glad you like it, Ty! Sure Hope you mean the edited videos.

https://github.com/TodePond/CellPond/assets/174252670/b81cefcf-bb4d-4447-8e4f-b39d6a3051f4
cant attach 2nd one. Should i send it on discord?

**Comment by Paalui** [comment]
I played around a bit more and came up with some new variations/improvements.

With adding and subtracting colors it was possible use one rule instead of two in some places. So i only had to use 38 Rules.
[GameOfLifeASync2.json](https://github.com/user-attachments/files/16117586/GameOfLifeASync2.json)
[GameOfLifeSync2.json](https://github.com/user-attachments/files/16117587/GameOfLifeSync2.json)

I also wanted to try counting the adjacent cells in some way instead of adding one rule for each combination of neighbors like I did before. For that i had to use several cells to represent on cell in the game, but it only needs 17 rules. 
[GameOfLifeSync3.json](https://github.com/user-attachments/files/16117588/GameOfLifeSync3.json)

To get around #353 scaled the grid up by 3 like this:
<img src="https://github.com/TodePond/CellPond/assets/174252670/0c8632ea-47b0-4ab4-89fe-dc928bebc059" height="350" />
[SetupX3.json](https://github.com/user-attachments/files/16117589/SetupX3.json)

**Reply by TodePond** [reply]
this is incredible. I'm on holiday at the moment but hope to try it all when I'm back

**Reply by TodePond** [reply]
also dang i really gotta fix that bug to enable more fun stuff like this

**Comment by AurySystem** [comment]
I did my own attempt at this today, and then I just realized github has discussions, so uhh.
[Cellpond life scanwaves.json](https://github.com/user-attachments/files/18762032/Cellpond.life.scanwaves.json)
![new cellpond life scanner](https://github.com/user-attachments/assets/8ec3dec3-0c48-4113-83ee-a4927285cbc7)
I used a scaning wiper to collapse frames and a 16 rule counter to check, which is a bit of a naive solution, but until I saw this featured in a talk and this discussion board it was the best I'd found yet. (and just generally uses up almost every channel for a calcuation)

**Reply by Paalui** [reply]
This is so smart. Do you use the left channel to count on what comparison you are and the middle for how many cells surround it?

**Reply by AurySystem** [reply]
Yeah! The left channel is storing which part of the comparison sequence it's on. The final value being used to store if it is applying rules to the count.
The middle channel is just how many live cells surround it. The final value is used to get it to store when the cells applying rules to the cell count are done to keep them in lock step with each other.
And then the right channel is storing the cell state. right before it applies rules it moves the value from 8 and 0 to 2 and 1 respectively.
I'm not a 100% sure why I did that, it's some sort of ready state before rule application, I'm not sure it's 100% necessary.

Otherwise there's a few other colors, but their use is setting up, and some of them might have ways to remove them.
999 for example sets up the screen scanning. (using 909 as well)
009 sets a live cell and clears it's the count to neighboors. (you actually see me mess up and place it close enough to the scanner that it freezes in the gif)

**Reply by TodePond** [reply]
this is so cool
